src/spec_coerce/core.cljc

```diff
@@ -182,11 +182,33 @@
        (URI. x)
        x)))
 
+(defn type->sym [x]
+  (cond (int? x)     `integer?
+        (float? x)   `float?
+        (boolean? x) `boolean?   ;; pointless but valid
+        ;;(symbol? x)  `symbol?  ;; doesn't work.
+        ;;(ident? x)   `ident?   ;; doesn't work.
+        (string? x)  `string?
+        (keyword? x) `keyword?
+        (uuid? x)    `uuid?
+        (nil? x)     `nil?       ;; even more pointless but stil valid
+
+        ;;#?(:clj (uri? x))     #?(:clj `uri?) ;; doesn't work.
+        #?(:clj (decimal? x)) #?(:clj `decimal?)))
+
+(defn spec-is-set? [x]
```

Comment:
a small thing, what you think about calling this `spec-is-homogeneous-set?`?

Reply:
Very logical. Done.